
registration_example:
  # stuff for the registration_client - if writing a custom client these lines are unnecessary
  topic:                  /laser_stitcher/full_scan #camera/depth/points
  load_cloud:             true
  initial_pause:          3.0
  betwixt_pause:          8.0
  bag_name_1:             /home/conor/ros_data/Fake_Walls/Segmented/screwy/5dps.bag
  bag_name_2:             /home/conor/ros_data/Fake_Walls/Segmented/screwy/10dps.bag
  introduce_error:        true
  introduced_error:       [0.10, -0.0, 0.0, 0.0, 0.0, 0.0]
  bag_name:               registered_cloud
  bag_topic:              /target_wall
  # registration stuff:
  transform_raw_clouds:             false
  transform_feature_clouds:         false
  transform_postprocessed_clouds:   false
  # registration algorithm parameters
  point_type:             XYZI    # XYZ, XYZN, XYZRGB, XYZRGBN, XYZI, XYZNI
  interest_point_parameters: 
    interest_point_type:  NONE    # NONE
  feature_parameters: 
    feature_type:         XYZNI   # XYZ, XYZN, XYZRGB, XYZRGBN
    ksearch:              30
    epsilon:              0.01
    max_iterations:       35
    step_size:            0.1
    resolution:           0.5
  correspondence_parameters:
    correspondence_type:  NONE    # NONE
    ransac_rejection:     false
    inlier_threshold:     0.02
    ransac_iterations:    10000
    temp:                 1
  transform_parameters:
    transformation_type:  ICP     # ICP, ICPNL, NDT
    epsilon:              0.0005
    max_iterations:       35
    max_dist:             0.2   
    euclidian_epsilon:    0.002 
    alpha:                [1.0, 1.0, 1.0, 1.0]
  # preprocessing stuff 
  preprocess:
    min_cloud_size:       50
    # NOTE - task names in task_list below MUST match the subheadings under the tasks heading
    task_list:            [ clip, voxelize ]
    tasks:
      to_base:
        type:             transform
        map_name:         base
      clip: 
        type:             clipping
        box:              [ -15, 15, -15, 15, -5, 12, 0, 0, 0, 0, 0, 0 ]
        keep_ordered:     true
      voxelize:           
        type:             voxelizing
        leaf_sizes:       [ 0.05, 0.05, 0.05 ]
        keep_ordered:     false
      to_map:
        type:             transform
        map_name:         map
  # postprocessing stuff
  postprocess:
    min_cloud_size:       50
    # NOTE - task names in task_list below MUST match the subheadings under the tasks heading
    task_list:            [ clip, voxelize ]
    tasks:
      clip: 
        type:             clipping
        box:              [-100, 100, -100, 100, -100, 100, 0, 0, 0, 0, 0, 0] #[ -6, 6, -2.1, 2.1, -1, 3.7, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
        keep_ordered:     true
      voxelize:           
        type:             voxelizing
        leaf_sizes:       [ 0.0001, 0.0001, 0.0001 ]
        keep_ordered:     false